### PR TITLE
[Refactoring] refactor subscribable

### PR DESCRIPTION
**Pull request by oa-s**

### CODE REVIEW

**Comment by oa-s** [comment]
@hboon  looks good for me, but could u please check on your side, handling for subscribable became more flexible, and easier with publishers

**Review by hboon — APPROVED** [review]
Seems OK otherwise. Thanks!

File: AlphaWallet/Common/Views/ViewModels/TokenCardRowViewModel.swift [hunk 1]
```diff
@@ -70,68 +71,31 @@ struct TokenCardRowViewModel: TokenCardRowViewModelProtocol {
...
-                            locality: value,
-                            state: tokenHolder.values.stateSubscribableStringValue,
-                            country: tokenHolder.values.countryStringValue
-                    )
-                }
-            }
-        }
+        let state = (tokenHolder.values.stateSubscribableValue?.publisher ?? .empty())
+            .map { $0?.stringValue }
+            .replaceEmpty(with: nil)
```
[the remark this answers is not in the document]
Comment (oa-s) [reply AlphaWallet/Common/Views/ViewModels/TokenCardRowViewModel.swift:88]: guess it should be ok without `.replaceEmpty(with:)`, updating

File: modules/AlphaWalletFoundation/AlphaWalletFoundation/Foundation/Subscribable.swift [hunk 2]
```diff
@@ -1,79 +1,44 @@
...
         return lhs.uuid == rhs.uuid
     }
 
-    public struct SubscribableKey: Hashable {
-        let id = UUID()
-
-        public static func == (lhs: SubscribableKey, rhs: SubscribableKey) -> Bool {
-            return lhs.id == rhs.id
-        }
+    private let subject: CurrentValueSubject<T?, Never>
```
[the remark this answers is not in the document]
Comment (oa-s) [reply modules/AlphaWalletFoundation/AlphaWalletFoundation/Foundation/Subscribable.swift:12]: its not easy to do it in one pr, easier to make it step by step

[the remark this answers is not in the document]
Comment (hboon) [reply modules/AlphaWalletFoundation/AlphaWalletFoundation/Foundation/Subscribable.swift:17]: Ok, yeah, that's good :)

**Pull request merged**